fuzzamoto-nyx-sys/src/nyx-agent.c

```diff
@@ -136,16 +136,31 @@ void nyx_dump_file_to_host(const char *file_name, size_t file_name_len,
 }
 
 void nyx_println(const char *message, size_t message_len) {
-    // Clamp to HPRINTF_MAX_SIZE - 1 to leave room for '\0'
-    size_t size = message_len;
-    if (size >= HPRINTF_MAX_SIZE) {
-      // Subtract one to accomodate the line ending
-      size = HPRINTF_MAX_SIZE - 1;
-    }
-    char message_copy[HPRINTF_MAX_SIZE];
-    memset(message_copy, 0, sizeof(message_copy));
-    memcpy(message_copy, message, size);
-    hprintf("%s\n", message_copy);
+  // case 1; our testcase is smaller than the nyx limitation
+  if (message_len < HPRINTF_MAX_SIZE) {
+      char message_copy[HPRINTF_MAX_SIZE];
```

Comment:
The changes to `nyx_println` should go into the second commit (95489d0b2bace24abd4a753bfee9218f7566312d)